### PR TITLE
Changes to support running controller and broker on same Node

### PR DESCRIPTION
**Pull request by anuj-apdev**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

**Test Configuration**:

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] Any variable changes have been validated to be backwards compatible

### CODE REVIEW

File: molecule/broker-scale-up/converge.yml [hunk 1]
```diff
@@ -25,11 +25,31 @@
...
-      # Kraft:TODO: get cluster ID from different method for kraft based scenarios
       shell: >
           /usr/bin/zookeeper-shell zookeeper1:2182  -zk-tls-config-file  /etc/kafka/zookeeper-tls-client.properties ls /brokers/ids
       register: zk_output
       when: not ('kafka_controller' in groups.keys() and groups['kafka_controller'] | length > 0)
+
+    - name: Get broker ID's
+      shell: >
+          kafka-metadata-quorum --command-config /etc/kafka/client.properties --bootstrap-server kafka-broker1:9091 describe --status | grep CurrentObservers | awk '{print $NF}'
+      register: zk_output
```
Comment (mansisinha) [inline molecule/broker-scale-up/converge.yml:36]: Shouldn't we name it like controller_output?

File: roles/kafka_controller/defaults/main.yml [hunk 2]
```diff
@@ -35,7 +35,7 @@ kafka_controller_final_java_args: "{{ kafka_controller_java_args + [ kafka_contr
 kafka_controller_service_overrides:
   User: "{{ kafka_controller_user if kafka_controller_user != kafka_controller_default_user else '' }}"
   Group: "{{ kafka_controller_group if kafka_controller_group != kafka_controller_default_group else '' }}"
-  ExecStart: "{% if installation_method == 'archive' %}{{ kafka_controller.server_start_file }} {{ kafka_controller.config_file }}{% endif %}"
+  ExecStart: "{{ kafka_controller.server_start_file }} {{ kafka_controller.config_file }}"
```
Comment (mansisinha) [inline roles/kafka_controller/defaults/main.yml:38]: won't this affect any archive scenario?

File: roles/variables/vars/main.yml [hunk 3]
```diff
@@ -101,11 +101,11 @@ zookeeper_combined_properties: "{{ zookeeper_properties | confluent.platform.com
 zookeeper_final_properties: "{{ zookeeper_combined_properties | combine(zookeeper_custom_properties) }}"
 
 #### Kafka controller variables
-kafka_controller_service_name: "{{ 'confluent-server' if confluent_server_enabled|bool else 'confluent-kafka'}}"
+kafka_controller_service_name: "confluent-kcontroller"
```
Comment (mansisinha) [inline roles/variables/vars/main.yml:104]: why are we not keeping a separate name for community kafka?

**Review by mansisinha — APPROVED** [review]

File: molecule/broker-scale-up/converge.yml (hunk 1 above)
Comment (anuj-apdev) [reply molecule/broker-scale-up/converge.yml:36]: i'll fix this in next PR.

File: roles/kafka_controller/defaults/main.yml (hunk 2 above)
Comment (anuj-apdev) [reply roles/kafka_controller/defaults/main.yml:38]: Hopefully not. I'll run a test.

File: roles/variables/vars/main.yml (hunk 3 above)
Comment (anuj-apdev) [reply roles/variables/vars/main.yml:104]: for the broker, we are still doing the same separation meaning different names on the basis of whether it's a community kafka or confluent Kafka. I don't see a reason to have an other name for the controller though.
let's discuss more on this in our next tech discussion session.

**Pull request merged**